algorithms.py

```diff
@@ -486,14 +493,17 @@ def increment_interval(self, left, right):
             if curr_interval.left == left:
                 if curr_interval.right <= right:
                     curr_interval.node += 1
+                    curr_interval.ancestral_to += ancestral_to
                     left = curr_interval.right
                     curr_interval = curr_interval.next
                 else:
                     self._split(curr_interval, right)
                     curr_interval.node += 1
+                    curr_interval.ancestral_to += ancestral_to
                     break
             else:
-                if curr_interval.right < left:
+                # verify this changed < to <= !!!!
```

Comment:
This doesn't seem correct - why did we change from < to <=?

Reply:
When using $<$ instead of $\leq$ the `OverlapCounter` makes segments of lengths 0 for breakpoints. For example, this is the overlap counter (see last line) that gets constructed for the following set of lineages:
`[0.0, 100.0: 0, 1]
[0.0, 100.0: 1, 1]
[0.0, 63.77461897661405: 2, 1]
[0.0, 100.0: 3, 1]
[0.0, 100.0: 4, 1]
[0.0, 100.0: 5, 1]
[0.0, 100.0: 6, 1]
[0.0, 100.0: 7, 1]
[0.0, 100.0: 8, 1]
[0.0, 100.0: 9, 1]
[63.77461897661405, 100.0: 2, 1]
overlap_counter:
[0, 63.77461897661405: 10, 10], [63.77461897661405, 63.77461897661405: 11, 11], [63.77461897661405, 100.0: 10, 10]
`

Reply:
aha, good spot!